### DISCUSSION TITLE
How to tell if SETNX put a value

### DISCUSSION THREAD

**Discussion by JeffBNimble** [redis/rueidis#347] [Q&A] [answered]
I'm using the following command to do a "set if not exists", but I'm not sure how to tell if the command set the value or not.

`
cmd := c.Do(context.Background(), c.B().Set().Key("someKey").Value(string("someValue")).Nx().Ex(expiration).Build())
`

What function do I send to cmd to know the result of the SETNX?

**Comment by rueian** [comment, marked as answer]
Hi @JeffBNimble,

According to [this](https://redis.io/commands/set/):
<img width="702" alt="image" src="https://github.com/redis/rueidis/assets/2727535/c016bb79-1c31-4808-818f-b99fc378460e">

You can check its return like this:

```go
err := c.Do(context.Background(), c.B().Set().Key("someKey").Value(string("someValue")).Nx().Ex(expiration).Build()).Error()
if rueidis.IsRedisNil(err) {
	// the SET operation was not performed because the NX condition was not met.
}
if err != nil {
	// other error
}
```
